lib/process.dart

```diff
@@ -38,6 +39,33 @@ class ProcessRunner {
 
   ProcessRunner(this.command, {this.args, this.cwd, this.env});
 
+  factory ProcessRunner.underShell(String command,
+      {List<String> args, String cwd, Map<String, String> env}) {
+    // This shouldn't be trusted for security.
+    final RegExp shellEscape = new RegExp('(["\'| \\\$!\\(\\)\\[\\]])');
+
+    final String shell = utils.env('SHELL');
+
+    if (shell == null) {
+      _logger.warning("Couldn't identify the user's shell");
+      return new ProcessRunner(command, args: args, cwd: cwd, env: env);
+    }
+
+    if (args != null) {
+      // Escape the arguments:
+      List<String> escaped = args.map((String arg) {
```

Comment:
You should declare this as `Iterable<String>` to avoid problems if this ever ran in checked mode :)